lib/srv/desktop/rdp/rdpclient/src/network_client.rs

```diff
@@ -0,0 +1,154 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use std::future::Future;
+use std::net::{IpAddr, Ipv4Addr};
+use std::pin::Pin;
+
+use ironrdp_connector::sspi::generator::NetworkRequest;
+use ironrdp_connector::sspi::network_client::NetworkProtocol;
+use ironrdp_connector::{custom_err, ConnectorResult};
+use ironrdp_tokio::AsyncNetworkClient;
+use reqwest::Client;
+use sspi::{Error, ErrorKind};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::{TcpStream, UdpSocket};
+use url::Url;
+
+pub(crate) struct ReqwestNetworkClient {
+    client: Option<Client>,
+}
+
+impl AsyncNetworkClient for ReqwestNetworkClient {
+    fn send<'a>(
+        &'a mut self,
+        request: &'a NetworkRequest,
+    ) -> Pin<Box<dyn Future<Output = ConnectorResult<Vec<u8>>> + 'a>> {
+        Box::pin(async move {
+            match &request.protocol {
+                NetworkProtocol::Tcp => self.send_tcp(&request.url, &request.data).await,
+                NetworkProtocol::Udp => self.send_udp(&request.url, &request.data).await,
+                NetworkProtocol::Http | NetworkProtocol::Https => {
+                    self.send_http(&request.url, &request.data).await
```

Comment:
Do we ever need this client to send HTTP data? Would it be better to error out or do nothing in this code path?

Reply:
No we don't. I simplify the client, now we error out for HTTP, HTTPS and UDP